appcelerator-archive/amp
# Cluster deployment on AWS

# content

- cluster deployment on AWS
- bootstrap of 3 Swarm manager nodes
- deployment and monitoring of worker nodes with InfraKit

# verification

- you should have aws credentials set in your home directory (~/.aws/credentials)
- add a terraform.tfvars in the bootstrap/terraform-aws folder with a content similar to
```
aws_name = "TAG"
aws_region = "us-west-2"
bootstrap_key_name = "YOUR-AWS-KEY-us-west-2"
aws_profile = "default"
infrakit_config_base_url = "https://raw.githubusercontent.com/appcelerator/amp/ikt-terraform-aws/bootstrap"
```
The key name should exist in AWS EC2 in this region and you should have the private key in your possession.
- deploy the bootstrap: ```docker run -v $HOME/.aws:/root/.aws:ro -v $PWD/bootstrap/terraform-aws:/data --workdir=/data hashicorp/terraform apply```. Take note of the IP displayed on the last line of the output.
- TBD: amp deployment
- log on the AWS console and look for the new instances in the EC2 page. Filter based on tag `Project` with the value defined in the file above. You should see 3 managers and 2 workers with random names.
- connect to the first manager (IP displayed above): ```ssh -i pem_file_path ubuntu@$IP```
- get the name of the worker group: ```docker exec infrakit infrakit group ls```
- destroy the worker group: ```docker exec infrakit infrakit group destroy workers-group-name```
- destroy the cluster: ```docker run -it -v $HOME/.aws:/root/.aws:ro -v $PWD/bootstrap/terraform-aws:/data --workdir=/data hashicorp/terraform destroy```
